src/ui/common/src/components/workflows/WorkflowSettings.tsx

```diff
@@ -629,33 +635,44 @@ const WorkflowSettings: React.FC<WorkflowSettingsProps> = ({
           object deletion.
         </Typography>
 
-        <List dense={false}>
+        <List style={{ maxHeight: '200px', overflow: 'auto' }} dense={true}>
           {Object.entries(deleteWorkflowResults)
             .map(([integrationName, objectResults]) =>
               objectResults.map((objectResult) => (
-                <ListItem key={`${integrationName}-${objectResult.name}`}>
-                  <ListItemIcon style={{ minWidth: '30px' }}>
-                    {objectResult.succeeded ? (
-                      <FontAwesomeIcon
-                        icon={faCircleCheck}
-                        style={{
-                          color: theme.palette.green[500],
-                        }}
-                      />
-                    ) : (
-                      <FontAwesomeIcon
-                        icon={faCircleXmark}
-                        style={{
-                          color: theme.palette.red[500],
-                        }}
-                      />
-                    )}
-                  </ListItemIcon>
-                  <ListItemText
-                    primary={displayObject(integrationName, objectResult.name)}
-                    secondary={<>{objectResult.succeeded}</>}
-                  />
-                </ListItem>
+                <>
+                  <ListItem key={`${integrationName}-${objectResult.name}`}>
+                    <ListItemIcon style={{ minWidth: '30px' }}>
+                      {objectResult.exec_state.status ===
+                      ExecutionStatus.Succeeded ? (
+                        <FontAwesomeIcon
+                          icon={faCircleCheck}
+                          style={{
+                            color: theme.palette.green[500],
+                          }}
+                        />
+                      ) : (
+                        <FontAwesomeIcon
+                          icon={faCircleXmark}
+                          style={{
+                            color: theme.palette.red[500],
+                          }}
+                        />
+                      )}
+                    </ListItemIcon>
+                    <ListItemText
+                      primary={displayObject(
+                        integrationName,
+                        objectResult.name
+                      )}
+                    />
+                  </ListItem>
+                  {objectResult.exec_state.status ===
+                    ExecutionStatus.Failed && (
+                    <Alert severity="error">
```

Comment:
Let's also use an `AlertTitle` inside this `Alert`. It'll make the formatting a little easier to read.